src/service/resources-impl.js

```diff
@@ -1012,19 +1012,40 @@ export class Resources {
     return this.vsync_.runPromise({
       measure: measurer || undefined,
       mutate: () => {
+        // TODO: Audit this system. Did this cause a layer invalidation (new
+        // layer, or removal of old layer)? Right now, we're only dirtying
+        // the measurements.
         mutator();
-
-        // TODO(jridgewell): This API needs to be audited. Common practice is
-        // to pass the amp-element in as the root even though we are only
-        // mutating children. If the amp-element is passed, we invalidate
-        // everything in the parent layer above it, where only invalidating the
-        // amp-element was necessary (only children were mutated, only
-        // amp-element's scroll box is affected).
-        this.layers_.dirty(element);
+        this.dirtyElement(element);
       },
     });
   }
 
+  /**
+   * Dirties the cached element measurements after a mutation occurs.
+   *
+   * TODO(jridgewell): This API needs to be audited. Common practice is
```

Comment:
Is there currently a way to tell layers that we're only mutating children of an element and not necessarily the entire element itself?

Reply:
You pass in the element being mutated.

```js

this.mutateElement(() => {
  // do something to child.
}, child);
```